core/controllers/reader.py

```diff
@@ -329,12 +329,19 @@ def post(self, unused_exploration_id):
         """
         # A domain object representing the old state.
         old_state = exp_domain.State.from_dict(self.payload.get('old_state'))
+        # The id of the exploration to which the state belongs.
+        exp_id = self.payload.get('exp_id')
```

Comment:
Actually, I am confused about this. Why don't we just use the exploration_id that's passed to the handler?